packages/web3-eth/src/eth_tx.ts

```diff
@@ -0,0 +1,326 @@
+import { EthExecutionAPI } from 'web3-common';
+import { Web3Context } from 'web3-core';
+import {
+	BlockTags,
+	convertToValidType,
+	HexString,
+	Numbers,
+	toHex,
+	ValidReturnTypes,
+	ValidTypes,
+} from 'web3-utils';
+import { privateKeyToAddress } from 'web3-eth-accounts';
+
+import Web3Eth from '.';
+import {
+	ChainIdMismatchError,
+	CommonOrChainAndHardforkError,
+	Eip1559GasPriceError,
+	Eip1559NotSupportedError,
+	InvalidGasOrGasPrice,
+	InvalidMaxPriorityFeePerGasOrMaxFeePerGas,
+	InvalidNonceOrChainIdError,
+	InvalidTransactionObjectError,
+	MissingChainOrHardforkError,
+	MissingCustomChainError,
+	MissingCustomChainIdError,
+	MissingGasError,
+	UnableToPopulateNonceError,
+	UnsupportedFeeMarketError,
+	UnsupportedTransactionTypeError,
+} from './errors';
+import { PopulatedUnsignedTransaction, Transaction } from './types';
+
+export function formatTransaction<
+	DesiredType extends ValidTypes = ValidTypes,
+	ReturnType = ValidReturnTypes[DesiredType],
+>(
+	transaction: Transaction,
+	desiredType: DesiredType,
+	overrideMethod?: (transaction: Transaction) => Transaction<ReturnType>,
+): Transaction<ReturnType> {
+	if (overrideMethod !== undefined) return overrideMethod(transaction);
+	const formattedTransaction = {
+		...transaction,
+		value:
+			transaction.value === '0x' ? '0x' : convertToValidType(transaction.value, desiredType),
+		gas: convertToValidType(transaction.gas, desiredType),
+		gasPrice: convertToValidType(transaction.gasPrice, desiredType),
+		type: convertToValidType(transaction.type, desiredType),
+		maxFeePerGas: convertToValidType(transaction.maxFeePerGas, desiredType),
+		maxPriorityFeePerGas: convertToValidType(transaction.maxPriorityFeePerGas, desiredType),
+		nonce: convertToValidType(transaction.nonce, desiredType),
+		chainId: convertToValidType(transaction.chainId, desiredType),
+		gasLimit: convertToValidType(transaction.gasLimit, desiredType),
+		v: convertToValidType(transaction.v, desiredType),
+		common: {
+			...transaction.common,
+			customChain: {
+				...transaction.common?.customChain,
+				networkId: convertToValidType(
+					transaction.common?.customChain?.networkId,
+					desiredType,
+				),
+				chainId: convertToValidType(transaction.common?.customChain?.chainId, desiredType),
+			},
+		},
+	};
+	// TODO - TSC is complaining that ReturnType could be instantiated with an
+	// arbitrary type which could be unrelated to 'string | number | bigint | undefined'
+	return formattedTransaction as unknown as Transaction<ReturnType>;
```

Comment:
@nazarhussain Do you have any idea on how to fix this? I thought `DesiredType extends ValidTypes` would fix this, but maybe I'm missing something?

Reply:
Assign the variable earlier with a proper type then you don't need to override here. This will also help to identify which attributes its complaining about. 

```ts
const formattedTransaction: Transaction<ReturnType> = {
```